File: Baekjun/24.8.2/8983. 사냥꾼/solution.py
from bisect import bisect_left

M, N, L = map(int, input().split())

A = sorted(map(int, input().split()))
ans = 0
for _ in range(N):
    x, y = map(int, input().split())
    idx = bisect_left(A, x)
    idxn = idx - 1

    if idx < M and (abs(x - A[idx]) + y) <= L:
        ans += 1
        continue

    if idxn >= 0:
        ans += int((abs(x - A[idxn]) + y) <= L)

print(ans)

File: Baekjun/24.8.2/8983. 사냥꾼/solution.rs
use std::io;
use std::io::BufRead;

fn main() {
    let stdin = io::stdin();
    let mut iterator = stdin.lock().lines();

    let mnl = iterator.next().unwrap().unwrap();
    let mut mnl_iter = mnl.split_whitespace();
    let m: usize = mnl_iter.next().unwrap().parse().unwrap();
    let n: usize = mnl_iter.next().unwrap().parse().unwrap();
    let l: i32 = mnl_iter.next().unwrap().parse().unwrap();

    let mut a: Vec<i32> = iterator.next().unwrap().unwrap()
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();
    a.sort();

    let mut ans = 0;

    for _ in 0..n {
        let line = iterator.next().unwrap().unwrap();
        let mut iter = line.split_whitespace();
        let x: i32 = iter.next().unwrap().parse().unwrap();
        let y: i32 = iter.next().unwrap().parse().unwrap();

        match a.binary_search(&x) {
            Ok(idx) => {
                if (x - a[idx]).abs() + y <= l {
                    ans += 1;
                }
            }
            Err(idx) => {
                if idx < m && (a[idx] - x).abs() + y <= l {
                    ans += 1;
                    continue;
                }
                if idx > 0 && (a[idx - 1] - x).abs() + y <= l {
                    ans += 1;
                }
            }
        }
    }

    println!("{}", ans);
}
